src/observable.rs

```diff
@@ -0,0 +1,75 @@
+use crate::Sealed;
+use core::marker::PhantomData;
+
+pub trait IntoObservationToken: Sized + crate::Sealed {
+    type Peripheral;
+    fn into_ot(self) -> ObservationToken<Self::Peripheral>;
+}
```

Comment:
I had to do this since we can not do this the trait impl rules

```rust
impl<T: Observable> Into<ObservationToken<T>> for T {
    fn into(self) -> ObservationToken<T> {
        ObservationToken {
            _p: PhantomData
        }
    }
}
```

do you know if there is a better way?

Reply:
The most recent version of this concept (with a few name changes and being used in a brand new context) is [here](https://github.com/AdinAck/proto-hal/blob/main/proto-hal/src/stasis.rs). The simple solution to this is to implement `From` rather than `Into`. This is a good idea in general (save some rare edge cases).

```rust
impl<P> From<P> for ObservationToken<P>
where
    P: Observable,
{
    fn from(_: P) -> Self {
        Self { _p: PhantomData }
    }
}
```

Another option - which eliminates the need for calling `.into()` to directly pass a peripheral as an observation token - is to create a new trait called `ObservationLock` which is implemented by all `P: Observable` and `ObservationToken`s. Interfaces would thus hold `impl ObservationLock`s rather than concrete `ObservationToken`s. This is described in [this](https://gist.github.com/AdinAck/c5713baf8b92d7075e10a9e03591569a) gist.

The former is a solution I am putting to use right now in [proto-hal](https://github.com/AdinAck/proto-hal/tree/main), the latter I suspect is a better solution, but have not implemented quite yet.

Hope this helps!

Reply:
Maybe the `ObservationLock` trait is unneeded since the interfaces can accept an `Into<ObservationToken<P>P>`. Clever!

Reply:
If we want the user to be able to destruct the consumer of the `ObservationToken` how does this affect things?

If the user passed an actual owned peripheral than would it not be best to just **not** convert it and let the consumer store the peripheral as is until it is destructed?

In that case would `ObservationLock` make more sense since that better signals what we are doing?

Reply:
Ah yes! *That* was why `ObservationLock` was good. Since the peripherals would store an `impl ObservationLock`, the `release()` would return back whatever was originally passed in.

Reply:
> The former is a solution I am putting to use right now in [proto-hal](https://github.com/AdinAck/proto-hal/tree/main)

Very interesting!